qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -168,6 +168,8 @@ def __init__(
         var_names: Optional[List[str]] = None,
         x_data: Optional[np.ndarray] = None,
         y_data: Optional[np.ndarray] = None,
+        weighted_residuals: Optional[np.ndarray] = None,
+        residuals: Optional[np.ndarray] = None,
```

Comment:
See comment about type of `residuals` in `convert_lmfit_result`. I think there's some confusion about the default value of `residuals`, when there's no residual plotting. I think you intended that it would be `None` in this case, however what's really passed to it is an empty list (a Python list, not an array). If so then fix in `CurveAnalysis._run_curve_fit`. In addition, when residual plotting is turned on, there's some inconsistency about the type of `residudals`, which is `np.ndarray` in the type hint doc but `np.array` in `_run_curve_fit`.

Reply:
I wanted it to be `None` to emphasize that the residuals wasn't calculated. I added to the code a comment explaining this and changed the code where I create the residuals data accordingly:
```
# if `plot_residuals` is ``False`` I would like the `residuals_model` be None to emphasize it
# wasn't calculated.
residuals_model = [] if self.options.get("plot_residuals") else None
```
And I changed the type hint of `CurveFitResult.__init__` method to be:
`residuals: Optional[Union[np.array, List]] = None,` 
as suggested